pkg/configstore/store.go

```diff
@@ -0,0 +1,179 @@
+package configstore
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+// Store holds config data
+type Store struct {
+	// If you care about sanity, never write here, just read
+	Data  map[string]interface{}
+	Dir   string
+	Hints map[string]string
+	sync.Mutex
+
+	ListFiles  func(string) ([]string, error)
+	ParseFile  func(string, interface{}) error
+	FileExists func(string) (bool, error)
+}
+
+func listFiles(dirname string) ([]string, error) {
+	files, err := ioutil.ReadDir(dirname)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make([]string, len(files))
+	for idx, file := range files {
+		ret[idx] = file.Name()
+	}
+
+	return ret, nil
+}
+
+func parseFile(path string, value interface{}) error {
+	reader, err := os.Open(path)
+	defer reader.Close()
+	if err != nil {
+		return fmt.Errorf("Failed to open %s: %v", path, err)
+	}
+
+	return json.NewDecoder(reader).Decode(value)
+}
+
+func fileExists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// NewStore creates a new config store
+// `dir` is a directory from where data will be loaded
+// `hints` is a dictionary to find which file to load for
+// a given key
+func NewStore(dir string, hints map[string]string) *Store {
+	if hints == nil {
```

Comment:
what usage do you see hints being nil?

Reply:
it's nil by default because in most cases there is direct mapping between key and filename, eventually we can pick a better default value that accounts for most inconsistencies and in the end i'd like it to go away completely but that needs revamping paasta's config system